src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -174,25 +174,41 @@ public void Dispose()
             }
         }
 
-        private void OnTimer(object sender, ElapsedEventArgs e)
+        private async void OnTimer(object sender, ElapsedEventArgs e)
         {
-            HandleTimerEvent().Wait();
+            await HandleTimerEvent();
         }
 
         internal async Task HandleTimerEvent()
         {
-            if (_remainingInterval != TimeSpan.Zero)
-            {
-                // if we're in the middle of a long interval that exceeds
-                // Timer's max interval, continue the remaining interval w/o
-                // invoking the function
-                StartTimer(_remainingInterval);
-                return;
-            }
+            bool timerStarted = false;
 
-            await InvokeJobFunction(DateTime.Now, false);
+            try
+            {
+                if (_remainingInterval != TimeSpan.Zero)
+                {
+                    // if we're in the middle of a long interval that exceeds
+                    // Timer's max interval, continue the remaining interval w/o
+                    // invoking the function
+                    StartTimer(_remainingInterval);
+                    timerStarted = true;
+                    return;
+                }
 
-            StartTimer(DateTime.Now);
+                await InvokeJobFunction(DateTime.Now, false);
+            }
+            catch (Exception ex)
+            {
+                // ensure background exceptions don't stop the execution schedule
+                _logger.LogError(ex, "Error occurred during scheduled invocation.");
```

Comment:
Can you note the function name here? These can be hard to correlate if there's a bunch of timers in the app.

Reply:
Done, thanks